control-agent/text-to-speech/speech-detection.py

```diff
@@ -131,160 +150,230 @@ def QnA(question):
+    intentJson = json.loads(intent.intent_json)
+
+    if "'up'" in str(intentJson):
+        up_down = "up"
+    elif "'down'" in str(intentJson):
+        up_down = "down"
+    if "'in'" in str(intentJson):
+        in_out = "in"
+    elif "'out'" in str(intentJson):
+        in_out = "out"
+
+    os.system(
+        "python /home/hello-robot/Chatbot/movearm.py '"
+        + up_down
+        + "' 1 '"
+        + in_out
+        + "' 1"
+    )
+
+
+def wrist_intent(intent):
+    """Intent response to move the stretch wrist"""
+    intentJson = json.loads(intent.intent_json)
+
+    try:
+        for entity in intentJson["entities"]:
+            if entity["type"] == "builtin.number":
+                amount = str(entity["resolution"]["value"])
+
+        os.system("python /home/hello-robot/Chatbot/moveWrist.py " + str(amount))
 
     except:
+        speech.speak("Move it where?")
+
+
+def vision_intent():
+    """Intent response to take a picture and analyse the contents
+    using Azure computer vision services"""
+    speech.speak("I'm looking")
+
+    image = realsense.take_photo()
+
+    result = speech.recognize(image)
+
+    if (
+        "person"
+        or "boy"
+        or "man"
+        or "woman" in result["description"]["captions"][0]["text"]
+    ):
+        # TODO more than one person?
+        try:
+            person = speech.identify_face(speech.recognize_face(image)[0]["faceId"])
+        except:
+            person = "someone I don't recognise"
+
+        speech.speak(
+            "I can see"
+            + str(result["description"]["captions"][0]["text"])
+            .replace("a person", str(person))
+            .replace("a man", str(person))
+            .replace("a boy", str(person))
+        )
+
+    else:
+        speech.speak("I can see" + str(result["description"]["captions"][0]["text"]))
+
+
+def intent_handler(intent):
+    """Handles the intent responces and calls the associated fucntions"""
+
+    if isinstance(intent, str):
         if intent != "No good match found in KB.":
             speech.speak(intent)
+            return
         else:
             speech.speak("I'm not sure I understood that")
+            return
+
+    if intent.intent_id == "Vision":
+        vision_intent()
+    elif intent.intent_id == "Move":
+        move_intent(intent)
+    elif intent.intent_id == "Weather.QueryWeather":
+        weather_intent()
+    elif intent.intent_id == "Time":
+        time_intent()
+    elif intent.intent_id == "Calibrate":
+        calibrate_intent()
+    elif intent.intent_id == "Stop":
+        return False
+    elif intent.intent_id == "Selfie":
+        selfie_intent()
+    elif intent.intent_id == "Grip":
+        grip_intent(intent)
+    elif intent.intent_id == "Arm":
+        arm_intent(intent)
+    elif intent.intent_id == "Wrist":
+        wrist_intent()
+
+    return True
+
+
+# Continuous loop starts here
+run = True
+# warm up confirmation
+speech.speak("starting up")
+
+while run == True:
```

Comment:
Will there be a shutdown intent @FergusKidd

Reply:
There is a 'Stop' intent which kills the process yes - a full safe shut down intent is an interesting idea though. It could ask to secure the clamp, confirm, and then shut down.